scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2200,11 +2200,14 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
       val rpPos = auto.endTokenPos
       @inline def addPos(body: Term): Term = {
         body match {
-          // For `Term.Ref`, avoid using autoPos to exclude parens tokens in Term.ApplyInfix.args.
+          // For `Term.Ref` & `Lit` other than Unit, avoid using autoPos
+          // to exclude parens tokens in Term.ApplyInfix.args.
           // For `a f (b)`, autoPos include `(b)` for ApplyInfix.args position.
           // https://github.com/scalacenter/scalafix/issues/1594
           case _: Term.Ref =>
             body
+          case l: Lit if l.value != () =>
```

Comment:
without the `if`, getting

```
[error] /home/runner/work/scalameta/scalameta/tests/shared/src/test/scala/scala/meta/tests/prettyprinters/SyntacticSuite.scala:1507:15: exception during macro expansion: 
[error] java.lang.RuntimeException: internal error: unpositioned prototype (): Lit.Unit(())
```

https://github.com/scalameta/scalameta/blob/60083a2d2eb4c23609694725b54d80bbe98dffe6/tests/shared/src/test/scala/scala/meta/tests/prettyprinters/SyntacticSuite.scala#L1469-L1474

Reply:
let's not look at the value but instead directly as Lit.Unit? In fact, instead of Ref and this, should you simply add parens for Lit.Unit and Term.Tuple only?

Reply:
another, possibly cleaner option, is to undo the recent Ref change here and instead remove addPos on line 2220 below, in the second branch of that match. elsewhere, the output of this method is unfolded, by mapping Lit.Unit back to Nil, and Term.Tuple to its arguments. however, the first branch of the match below returns a different expression than what addPos wraps there.